### PR TITLE
Add `Validate` method in `Func` interface to validate a kanister function

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview

This PR fundamentally just refactors the way we used to validate a blueprint phase. Initially when we added blueprint validation, we called `Validate` on a phase (`phase.go`) that would get arguments of the respective function and then check if the arguments are proper or not.
This PR just delegates this validation logic to the function itself, and the reason is
- If we are validating the functions, it's better to have the validation logic to closer to the function
- And, if there are some specific validation that needs to be done for just some functions that's easily achievable if validation is implemented this way. (for example if for some function we are allowed to provide labels and validate should also validate those labels)

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- fixes #issue-number

## Test Plan

```
~/opensource/kanister/pkg/blueprint/validate (validate-kanf*) » go test
Failed the 'validation of phase 00 in action backup' check.. ❌
Passed the 'validation of phase 10 in action backup' check.. ✅
Passed the 'validation of phase 11 in action backup' check.. ✅
Failed the 'validation of action backup' check.. ❌
Passed the 'validation of phase 30 in action backup' check.. ✅
Failed the 'validation of phase 40 in action backup' check.. ❌
Failed the 'validation of phase 51 in action backup' check.. ❌
Failed the 'validation of phase 61 in action backup' check.. ❌
Passed the 'validation of phase 71 in action backup' check.. ✅
Passed the 'validation of phase 70 in action backup' check.. ✅
{"FallbackVersion":"v0.0.0","File":"pkg/phase.go","Function":"PrepareData","Line":184,"PreferredVersion":"v0.0.1","level":"info","msg":"Falling back to default version of the function","time":"2024-08-05T12:09:06.408961+05:30"}
Passed the 'validation of phase 00 in action backup' check.. ✅
Passed the 'validation of phase 01 in action backup' check.. ✅
{"FallbackVersion":"v0.0.0","File":"pkg/phase.go","Function":"PrepareData","Line":184,"PreferredVersion":"v0.0.1","level":"info","msg":"Falling back to default version of the function","time":"2024-08-05T12:09:06.409274+05:30"}
Failed the 'validation of phase 10 in action backup' check.. ❌
Passed the 'validation of phase 20 in action backup' check.. ✅
Failed the 'validation of phase 21 in action backup' check.. ❌
{"FallbackVersion":"v0.0.0","File":"pkg/phase.go","Function":"PrepareData","Line":184,"PreferredVersion":"v0.0.1","level":"info","msg":"Falling back to default version of the function","time":"2024-08-05T12:09:06.409295+05:30"}
{"FallbackVersion":"v0.0.0","File":"pkg/phase.go","Function":"PrepareData","Line":184,"PreferredVersion":"v0.0.1","level":"info","msg":"Falling back to default version of the function","time":"2024-08-05T12:09:06.409301+05:30"}
Passed the 'validation of phase 30 in action backup' check.. ✅
Passed the 'validation of phase 31 in action backup' check.. ✅
OK: 3 passed
PASS
ok      github.com/kanisterio/kanister/pkg/blueprint/validate   0.899s
```

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -114,3 +115,21 @@ func RoundUpDuration(t time.Duration) time.Duration {
 	}
 	return t.Round(time.Hour)
 }
+
+func CheckRequiredArgs(reqArgs []string, args map[string]interface{}) error {
```
Comment (PrasadG193) [inline pkg/utils/utils.go:119]: Lets add comment on both the methods

File: pkg/function/delete_rds_snapshot.go [hunk 2]
```diff
@@ -114,10 +115,10 @@ func deleteRDSSnapshot(ctx context.Context, snapshotID string, profile *param.Pr
 	return nil, errors.Wrap(err, "Error waiting for Aurora DB cluster snapshot to be deleted")
 }
 
-func (crs *deleteRDSSnapshotFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+func (d *deleteRDSSnapshotFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
 	// Set progress percent
-	crs.progressPercent = progress.StartedPercent
-	defer func() { crs.progressPercent = progress.CompletedPercent }()
+	d.progressPercent = progress.StartedPercent
+	defer func() { d.progressPercent = progress.CompletedPercent }()
```
Comment (PrasadG193) [inline pkg/function/delete_rds_snapshot.go:121]: Thanks for doing this :)

**Review by PrasadG193 — APPROVED** [review]
LGTM

File: go.mod [hunk 3]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions v1.3.0
 	github.com/Masterminds/semver v1.5.0
 	github.com/Masterminds/sprig v2.22.0+incompatible
-	github.com/aws/aws-sdk-go v1.55.0
+	github.com/aws/aws-sdk-go v1.55.5
```
Comment (PrasadG193) [inline go.mod:26]: I don't think this is necessary for the changes we are making

Comment (viveksinghggits) [reply go.mod:26]: right, but let's merge this. `make release-snapshot` sometimes fails (at least on mac) if we don't have this change.

Comment (viveksinghggits) [reply go.mod]: I removed this change.

**Pull request merged**